tools/feature-flags/deploy/auth0/actions/Check GitHub teams/code.js

```diff
@@ -0,0 +1,55 @@
+const ALLOWED_TEAMS = [
+  "ministryofjustice:probation-integration",
+  "ministryofjustice:ndst"
+]
+
+/**
+ * Handler that will be called during the execution of a PostLogin flow.
+ *
+ * @param {Event} event - Details about the user and the context in which they are logging in.
+ * @param {PostLoginAPI} api - Interface whose methods can be used to change the behavior of the login.
+ */
+exports.onExecutePostLogin = async (event, api) => {
+  const _ = require('lodash');
+  const axios = require("axios");
+  const ManagementClient = require('auth0').ManagementClient;
+
+  // Apply to 'github' connections only
+  if (event.connection.strategy !== 'github') {
+    api.access.deny('User must be authenticated via GitHub');
+    return;
+  }
+
+  // Get user
+  try {
+    var authUser = await new ManagementClient({
+      domain: event.secrets.domain,
+      clientId: event.secrets.clientId,
+      clientSecret: event.secrets.clientSecret,
+    }).getUser({ id : event.user.user_id })
+  } catch (e) {
+    console.log(e);
+    api.access.deny('Failure to get Auth0 user data');
+    return;
+  }
+
+  // Get Github teams
+  const githubIdentity = _.find(authUser.identities, { connection: 'github' });
+  const githubResponse = await axios.get('https://api.github.com/user/teams', {
+    headers: { 'Authorization': `token ${githubIdentity.access_token}` }
+  });
+  if (githubResponse.status !== 200) {
+    console.log(githubResponse);
```

Comment:
For the errors, yes - it's really helpful to be able to see the cause for unexpected login failures in the auth0 dashboard.  For instance if our GitHub token was revoked or something, this helps us diagnose it.  Otherwise your login fails and you have no idea why.